### PR TITLE
Update RELEASE_PROCESS.rst

### PR DESCRIPTION
**Pull request by akrabat**
Various updates after releasing a new version to hopefully make it clearer next time.

### CODE REVIEW

File: doc/RELEASE_PROCESS.rst [hunk 1]
```diff
@@ -33,39 +33,72 @@ This is an outline of what needs to be done in order to release rst2pdf.
 
 #. Build manual
 
-   You will need the `White Rabbit`_ font installed. We check out the tag first so that the version
+   You will need the `White Rabbit`_ font installed. We check out the tag install via pip first so that the version
    on the first page of the PDF is labelled as final and then generate the HTMl and PDF docs.
```
Comment (lornajane) [inline doc/RELEASE_PROCESS.rst:37]:
```suggestion
   on the first page of the PDF is labelled as final and then generate the HTML and PDF docs.
```

File: doc/RELEASE_PROCESS.rst [hunk 2]
```diff
@@ -33,39 +33,72 @@ This is an outline of what needs to be done in order to release rst2pdf.
 
 #. Build manual
 
-   You will need the `White Rabbit`_ font installed. We check out the tag first so that the version
+   You will need the `White Rabbit`_ font installed. We check out the tag install via pip first so that the version
```
Comment (lornajane) [inline doc/RELEASE_PROCESS.rst:36]: I'm not sure this works. Is it "check out the tag AND install via pip" ? I think checking out the tag first links to the second half of the sentence which is why I'm confused.

File: doc/RELEASE_PROCESS.rst [hunk 3]
```diff
@@ -33,39 +33,72 @@ This is an outline of what needs to be done in order to release rst2pdf.
...
+    Double check that you do not have any changes in your working directory that are not committed. If you do, stash
+    them as otherwise uploading to PyPI will not work.
+
     ::
 
        $ python setup.py egg_info -b "rc1" sdist bdist_wheel
 
     If you're doing an alphaX, betaX or postX, then change ``-b "rc1"`` appropriately
 
+#. Setup PyPI if you haven't already
```
Comment (lornajane) [inline doc/RELEASE_PROCESS.rst:75]:
```suggestion
#. Set up PyPI if you haven't already
```

**Review by lornajane — COMMENTED** [review]
I'm sure this is am improvement but there's one bit that tripped me up, can you help to untangle the meaning(s)?

File: doc/RELEASE_PROCESS.rst (hunk 2 above)
Comment (akrabat) [reply doc/RELEASE_PROCESS.rst]: You're right. This is not good. I've taken another stab at it, including moving the font requirement to the top as that's where we mention the other dependencies required.

**Review by lornajane — APPROVED** [review]
Thanks for updating this for future-us :)

**Pull request merged**